ambry-router/src/main/java/com.github.ambry.router/DecryptJob.java

```diff
@@ -69,33 +89,27 @@ public void run() {
     Exception exception = null;
     ByteBuffer decryptedBlobContent = null;
     ByteBuffer decryptedUserMetadata = null;
-    try {
-      Object containerKey = kms.getKey(blobId.getAccountId(), blobId.getContainerId());
-      Object perBlobKey = cryptoService.decryptKey(encryptedPerBlobKey, containerKey);
-      if (encryptedBlobContent != null) {
-        decryptedBlobContent = cryptoService.decrypt(encryptedBlobContent, perBlobKey);
-      }
-      if (encryptedUserMetadata != null) {
-        decryptedUserMetadata = cryptoService.decrypt(encryptedUserMetadata, perBlobKey);
+    if (!isComplete()) {
+      try {
+        Object containerKey = kms.getKey(blobId.getAccountId(), blobId.getContainerId());
```

Comment:
Why are you using a generic Object here?

Reply:
We didn't have a need to make it type specific as we don't make any calls over the keys within CryptoJobs.